src/operations/command.ts

```diff
@@ -90,6 +99,12 @@ export abstract class CommandOperation<
     if (parent && parent.logger) {
       this.logger = parent.logger;
     }
+
+    // Assign BSON serialize options to OperationBase, preferring options over parent options.
+    // base accounts for the fact that Collection stores bson options in s, while other parents,
+    // like Db, stores bson options in s.options
```

Comment:
Maybe this is cause to change that? I wonder how much it would effect to make collection keep its bsonOptions in options, or have db spread them on to `s`.

That could make this into a lot of work, so total throwaway comment if so, but curious if you have insight on what that would entail?

Reply:
I don't think it would have a strong effect at all to move the Collection `bsonOptions` to `s.options`-- as far as I can tell, very few places were using the Collection `bsonOptions` on `s`, and I deleted most of them in this PR. Would be happy to make this change. 

As for spreading other parent `bsonOptions` onto `s`, `Db` was just one example. I think we would also have to change `cursor` too, and possibly others.

Reply:
I think that would be nice for consistency
I'm thinking now the original reason that collection might pick off its own copy of these options is to make sure that it doesn't edit its parent's options, I took a quick look around and it seems like collection does get its own separate options object but that could be a bug you run into if you give this a go, the options prop that collections uses shouldn't be the same reference to Db's options object.